keps/sig-security/2763-ambient-capabilities/README.md

```diff
@@ -0,0 +1,1075 @@
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+[Ambient capabilities](https://man7.org/linux/man-pages/man7/capabilities.7.html) is a set of capabilities that are preserved across an execve(2) of a program that is not privileged. This KEP proposes that kubernetes provide a way to set ambient capabilities for containers through the Pod manifest. It also proposes changes that must be made to `containerd` and `CRI-O` to enable ambient capabilities end-to-end.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Running containers as non-root has been a long recommended best-practice in kubernetes and we have published [blogs](https://kubernetes.io/blog/2018/07/18/11-ways-not-to-get-hacked/#8-run-containers-as-a-non-root-user) recommending this best practice. In addition to running as non-root it is also recommended that all capabilities other than the ones required are dropped from the container. Since most containers don’t require any capabilities this guidance becomes easy to follow.
+
+**The following works:-**
+```Dockerfile
+FROM ubuntu
+
+COPY main /bin/simpleserver
+```
+
+```bash
+docker build -t simpleserver:nofilecaps .
+```
+
+```yaml
+# pod.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-web
+  labels:
+    role: myrole
+spec:
+  containers:
+    - name: web
+      securityContext:
+              runAsUser: 1000
+              runAsGroup: 1000
+              allowPrivilegeEscalation: false
+              capabilities:
+                      drop: ["All"]
+      image: simpleserver:nofilecaps
+      command: ["simpleserver", "--port", "8080"]
+```
+
+```bash
+kubectl apply -f pod.yaml
+pod/static-web created
+
+kubectl get pods                                                        
+NAME         READY   STATUS    RESTARTS   AGE
+static-web   1/1     Running   0          10s
+```
+
+**The following does not work:** because the container is running as non-root user `1000` and since it is mounting a privileged port it will require `CAP_NET_BIND_SERVICE` linux capability.
+
+```yaml
+# pod.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-web
+  labels:
+    role: myrole
+spec:
+  containers:
+    - name: web
+      securityContext:
+              runAsUser: 1000
+              runAsGroup: 1000
+              allowPrivilegeEscalation: false
+              capabilities:
+                      drop: ["All"]
+                      add: ["NET_BIND_SERVICE"]
+      image: simpleserver:nofilecaps
+      command: ["simpleserver", "--port", "80"]
+```
+
+```bash
```

Comment:
nit: This isn't bash (it's commands mixed with sample output)

Reply:
Updated.